pandas/tests/indexes/datetimelike.py

```diff
@@ -75,3 +75,17 @@ def test_map_dictlike(self, mapper):
         expected = pd.Index([pd.NaT] * len(self.index))
         result = self.index.map(mapper([], []))
         tm.assert_index_equal(result, expected)
+
+    def test_asobject_deprecated(self):
+        d = pd.date_range('2010-01-1', periods=3)
```

Comment:
add the issue number as a comment

use ``d = self.create_index()``

Reply:
Ok, uploaded. ``self.create_index()`` takes care of calling the correct index type?